src/pages/costModels/createCostModelWizard/reviewWithDistribution.tsx

```diff
@@ -0,0 +1,136 @@
+          <Stack hasGutter>
+            <StackItem>
+              <Title headingLevel="h2" size="xl">
+                {t('cost_models_wizard.review.title_details')}
+              </Title>
+            </StackItem>
+            <StackItem>
+              <TextContent>
+                <Text component={TextVariants.h6}>
+                  <Trans i18nKey="cost_models_wizard.review.sub_title_details">
+                    <strong>{t('cost_models_wizard.review.create_button')}</strong>
+                    <strong>{t('cost_models_wizard.review.back_button')}</strong>
+                  </Trans>
+                </Text>
+              </TextContent>
+            </StackItem>
+            <StackItem>
+              <TextContent>
+                <TextList component={TextListVariants.dl}>
+                  <TextListItem component={TextListItemVariants.dt}>{t('name')}</TextListItem>
+                  <TextListItem component={TextListItemVariants.dd}>{name}</TextListItem>
+                  <TextListItem component={TextListItemVariants.dt}>{t('description')}</TextListItem>
+                  <TextListItem component={TextListItemVariants.dd}>{description}</TextListItem>
+                  {type === 'OCP' && (
+                    <>
+                      <TextListItem component={TextListItemVariants.dt}>
+                        {t('cost_models_wizard.steps.price_list')}
+                      </TextListItem>
+                      <TextListItem component={TextListItemVariants.dd}>
+                        {tiers.length > 0 ? <RateTable t={t} tiers={tiers} /> : t('cost_models_wizard.no_rates')}
+                      </TextListItem>
+                    </>
+                  )}
+                  <TextListItem component={TextListItemVariants.dt}>
+                    {t('cost_models_wizard.review.markup')}
+                  </TextListItem>
+                  <TextListItem component={TextListItemVariants.dd}>{isDiscount ? '-' + markup : markup}%</TextListItem>
+                  <TextListItem component={TextListItemVariants.dt}>
+                    {t('cost_models_details.distribution_type')}
+                  </TextListItem>
+                  <TextListItem component={TextListItemVariants.dd}>{distribution}</TextListItem>
+                  <TextListItem component={TextListItemVariants.dt}>
+                    {t('cost_models_wizard.review.sources')}{' '}
+                    {sources.find(src => src.selected && Boolean(src.costmodel)) && (
+                      <WarningIcon text={t('cost_models_wizard.warning_override_sources')} />
+                    )}
+                  </TextListItem>
+                  <TextListItem component={TextListItemVariants.dd}>
+                    {sources
+                      .filter(r => r.selected)
+                      .map(r => r.name)
+                      .join(', ')}
+                  </TextListItem>
+                </TextList>
```

Comment:
This looks like an exact copy of `review.tsx` with the exception of these new lines. It would be best to conditionally include this in the existing file.

Reply:
I agree that there is duplication of code, I was following the the structure that was already present using the source_type along with steps to decide on which component to load. I can certain rewrite the code to be conditional but seems like if we go that route that we might want to revisit how this wizard is put together and maybe do a larger refactor in another PR?

Reply:
I am using the same mechanism for these pages as well. I agree that there is dupe code here. The way it looks like it was design is to use the sources type as a switch and then based on that you have a series of steps. Those steps are components.  If you look at steps.tsx that is how pricelist and these new changes are handled.

Reply:
There isn't a separate review page for price list, tho. I'm not sure I follow the logic here

Reply:
Yes sorry about that I should have read this a bit clearer its the review step looking at that now.